xblock/core.py

```diff
@@ -290,6 +307,44 @@ def needs_serialization(self):
         return any([field.is_set_on(self) for field in self.fields.itervalues()])
 
 
+# -- UI-only Block
+class UIBlock(RuntimeServicesMixin, HandlersMixin, ViewsMixin, SharedBlockBase):
+    """Base class for UI Blocks.
+
+    This class differs from XBlocks in that they are intended only to
+    provide pluggable user interfaces. They explicitly have no state
+    and cannot be persisted.
```

Comment:
.. for now.

Eventually, I do think we'll want to add support for persistence and state for UIBlocks as well.